examples/ivr_outbound_dialer.py

```diff
@@ -0,0 +1,359 @@
+
+# Specify FreeSWITCH or Sangoma NSG IP information
+# In this example the sample app is running on
+# Sangoma NSG appliance hence the use of local address
+host = "127.0.0.1"
+port = 8821
+
+
+# IVR Call Logic
+# This is the main IVR logic
+# All custom development should be done here.
+# Variables
+#   self.<variables>       are global in nature.
+#   call.vars.['var_name'] should be used for per call info and state
+#
+class IVRCallLogic(object):
+
+    def prepost(self, client, listener):
+        # Get the install directory of FreeSWITCH or Sangoma NSG and append recording to it
+        # By default FreeSWITCH resides in /usr/local/freeswitch, NSG resides in /usr/local/nsg
+        self.base_dir = client.cmd('global_getvar base_dir')
+        self.recdir = self.base_dir + "/recording"
+        log.info("Setting recording dir to '{}" . format(self.recdir))
+
+        # Get the install directory of NSG and append sounds to it
+        self.sound_dir = self.base_dir +  "/sounds"
+        log.info("Setting sounds dir to '{}" . format(self.sound_dir))
+
+        self.stereo = False
+
+        # mod_sndfile module is a must in order to play prompts
+        # Example of how to execute FreeSWITCH/NSG commands as from the CLI
+        try:
+            client.cmd('load mod_sndfile')
+        except switchy.utils.CommandError:
+            pass
+
+        # Example of how to use a Timer to measure elapsed time
+        # This example does not use it, but its use is commented out and
+        # left for demonstration purposes
+        self.dtmf_timeout = switchy.utils.Timer()
+        self.dtmf_timeout_period = 3.0
+
+    @event_callback('CHANNEL_PARK')
+    def on_park(self, sess):
+        # Example of how to answer an inbound call
+        # This application demonstrates outbound dialing
+        if sess.is_inbound():
+            sess.answer()
+
+    @event_callback("CHANNEL_ANSWER")
+    def on_answer(self, sess):
+        call = sess.call
+
+        # This application does not deal with inbound calls.
+        if sess.is_inbound():
+            pass
+
+        # Outbound call hsa just been answered
+        # Developer would start the introductory IVR message
+        if sess.is_outbound():
+
+            # Start recording a call
+            call.vars['record'] = True
+            sess.start_record('{}/callee_{}.wav'.format(self.recdir, sess.uuid), stereo=self.stereo)
+
+            # Play IVR initial greeting
+            call.vars['play_welcome'] = True
+            play_filename = '{}/en/us/callie/ivr/8000/ivr-welcome.wav'.format(self.sound_dir)
+            sess.playback(play_filename)
+
+            # At this point we wait for PLAYBACK_STOP event to finish
+            # We could trigger a timeout as in DTMF section
+
+    # Timer handler function that implements DTMF timeout
+    def dtmf_timeout_action(self, sess):
+        call = sess.call
+
+        log.info("'{}': DTMF timeout".  format(sess.uuid))
```

Comment:
I'm surprised flake8 didn't catch that? Should be:
`log.info("'{}': DTMF timeout".format(sess.uuid))`